src/sections/upload-dataset-files/UploadDatasetFiles.tsx

```diff
@@ -1,19 +1,104 @@
-import { useEffect } from 'react'
+import { useEffect, useState } from 'react'
 import { FileRepository } from '../../files/domain/repositories/FileRepository'
 import { useLoading } from '../loading/LoadingContext'
 import { useDataset } from '../dataset/DatasetContext'
 import { PageNotFound } from '../page-not-found/PageNotFound'
 import { BreadcrumbsGenerator } from '../shared/hierarchy/BreadcrumbsGenerator'
+import { FileUploader } from './FileUploader'
+import { FileUploadTools } from '../../files/domain/models/FileUploadState'
+import { uploadFile } from '../../files/domain/useCases/uploadFile'
+import { useTranslation } from 'react-i18next'
 
 interface UploadDatasetFilesProps {
   fileRepository: FileRepository
 }
 
-export const UploadDatasetFiles = ({
-  fileRepository: _fileRepository
-}: UploadDatasetFilesProps) => {
+export const UploadDatasetFiles = ({ fileRepository: fileRepository }: UploadDatasetFilesProps) => {
+  const { t } = useTranslation('files')
   const { setIsLoading } = useLoading()
   const { dataset, isLoading } = useDataset()
+  const [fileUploaderState, setState] = useState(FileUploadTools.createNewState([]))
```

Comment:
Notice that you can use react useReducer when you have lots of states and logic going on.
That way you can keep the state synced in one place and read it from one place also.
Maybe you don't need it, just a comment :).

https://react.dev/reference/react/useReducer

Here is an example in the [selectMultipleReducer](https://github.com/IQSS/dataverse-frontend/blob/develop/packages/design-system/src/lib/components/select-multiple/selectMultipleReducer.ts)

Reply:
I did not know this. It looks very similar to what I did. It is my first time I did write some code with React, so it is hard to tell if the reducer would be better here. The function calls look more natural to me in the callbacks as I did in my code, and the reducer looks more natural for buttons, etc., but it might be just me, because I do not know React very well... I will leave it like it is for now, we might want to delete some of my code, or change how the state works, what is in it, etc.